### PR TITLE
Refurbish developing modules content - stage 1

### PR DESCRIPTION
**Pull request by dharmabumstead**
##### ISSUE TYPE

 - Docs Pull Request

##### ANSIBLE VERSION

```

```

##### SUMMARY

This is the initial stage of a refurbishment of the module development documentation that @gundalow and I have been working on. 

```

```

### CODE REVIEW

**Comment by gundalow** [comment]
Did you run your link checker to confirm there are no dead links. Does that tool also check from broken anchor links?

**Comment by dharmabumstead** [comment]
Yes. Links in that section are OK, other than the 'Edit on Github' links in the headers, which won't work until we merge this and push the changes to production.

**Comment by gundalow** [comment]
Excellent I'll have this reviewed for when you wake up.

File: docs/docsite/rst/dev_guide/developing_modules.rst [hunk 1]
```diff
@@ -14,876 +21,46 @@ See :doc:`../modules` for a list of existing modules.
...
 
-You can also rename modules and keep an alias to the old name by using a symlink that starts with _.
-This example allows the stat module to be called with fileinfo, making the following examples equivalent::
+:doc:`developing_modules_general`
+    A general overview of how to develop, debug, and test modules
+:doc:`developing_modules_documenting`
+    How to include in-line documentation in your module.
+:doc:`developing_modules_best_practices`
+    Best practices, recommendations, and things to avoid.
+:doc:`developing_modules_contributing`
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/developing_modules.rst]: Please rename this link to be `developing_modules_checklist`

File: docs/docsite/rst/dev_guide/developing_modules_contributing.rst [hunk 2]
```diff
@@ -0,0 +1,230 @@
+.. _module_contribution:
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/developing_modules_contributing.rst]: Please rename this file to be `developing_modules_checklist`

**Review by gundalow — CHANGES_REQUESTED** [review]

File: docs/docsite/rst/dev_guide/developing_modules.rst [hunk 3]
```diff
@@ -14,876 +21,46 @@ See :doc:`../modules` for a list of existing modules.
...
+There are a lot of existing modules available, and more that are in development. You should check out the list of existing modules at :doc:`../modules` or look at the `module PRs <https://github.com/ansible/ansible/labels/module>`_ for the ansible repository on Github to see if a module that does what you want exists or is in development.
 
-    ansible/hacking/test-module -m ./timetest.py -a "time=\"March 14 12:23\""
+2. Should you use or develop an action plugin instead? 
 
-This should return something like::
+Action plugins get run on the master instead of on the target. For modules like file/copy/template, some of the work needs to be done on the master before the module executes on the target. Action plugins execute first on the master and can then execute the normal module on the target if necessary. 
 
-    {"changed": true, "time": "2012-03-14 12:23:00.000307"}
+For more information about action plugins, go here: (TODO: link here)
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/developing_modules.rst]: Do we have a link for this yet, or should we just remove this?

**Review by gundalow — DISMISSED** [review]

**Comment by dharmabumstead** [comment]
Thanks @gundalow!

**Comment by gundalow** [comment]
Woot

**Pull request merged**